libs/contrib/Data/Nat/Factor.idr

```diff
@@ -358,7 +358,7 @@ notLteAndGt (S k) (S j) aLteB aGtB = notLteAndGt k j (fromLteSucc aLteB) (fromLt
 gcd_step : (x : Search) ->
     (rec : (y : Search) -> Smaller y x ->  (f : Nat ** GCD f (left y) (right y))) ->
     (f : Nat ** GCD f (left x) (right x))
-gcd_step (SearchArgs Z _ bLteA {bNonZero}) _ = absurd . succNotLTEzero $ lteTransitive bNonZero bLteA
+gcd_step (SearchArgs Z _ bLteA {bNonZero}) _ = absurd . succNotLTEzero $ transitive {ty = Nat} bNonZero bLteA
```

Comment:
@Sventimir Here's a case where an "implicit" argument must be specified explicitly. Strangely, _some_ argument must be specified, but it doesn't matter _which_ argument is specified.

This does not work:

```idris
transitive bNonZero bLteA
```

But the following all do work:

```idris
transitive {ty = Nat} bNonZero bLteA
```

```idris
transitive {rel = LTE} bNonZero bLteA
```

```idris
transitive {x = S Z} bNonZero bLteA
```

```idris
transitive {y = a} bNonZero bLteA
```

```idris
transitive {z = Z} bNonZero bLteA
```

So nothing in particular needs to be specified, but one thing or another does need to be specified. I don't know anything about the internals, but as a user it feels like the typechecker needs a little nudge in the right direction to get started. Hopefully inference will improve and these nudges won't be required.

Reply:
That's interesting. I'm guessing that's because of packing all the arguments to the `gcd_step` in this `Search` type. The compiler only sees this `Search` type and does not dig deep enough to understand that it's all about `Nat`s. Notice that from any of these arguments it can be inferred that we want `transitive` for `Nat`/`LTE`. So you're probably right in that the compiler needs just a little nudge.

Reply:
It's even more general than that. Every example of required implicit arguments in this PR acts the same way, including the simple call to `reflexive` in `Data.List.Views`.